float64 LowBound
float64 UpBound
